### PR TITLE
throw exception when .htaccess file is not writable

### LINKED ISSUE #32227: maintenance:update:htaccess does not show any error when writing of .htaccess fails
### Steps to reproduce
1. install owncloud in an sub-folder of an apache directory
2. run `sudo -u www-data ./occ config:system:set htaccess.RewriteBase --value /owncloud-core/`
3. make sure in config.php `'htaccess.RewriteBase' => '/owncloud-core/',` is set
4. make sure www-data can**not**  write the .htaccess file
5. run `sudo -u www-data ./occ maintenance:update:htaccess`

### Expected behaviour
Error shown saying that `.htaccess` file could not be updated

### Actual behaviour
output `.htaccess has been updated`

### Server configuration
**Operating system**:
Ubuntu 18.04
**Web server:**
Apache
**Database:**
Sqlite
**PHP version:**
7.2
**ownCloud version:** (see ownCloud admin page)
10.1
**Updated from an older ownCloud or fresh install:**
fresh
**Where did you install ownCloud from:**
git
**The content of config/config.php:**

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost"
        ],
        "overwrite.cli.url": "http:\/\/localhost\/owncloud-core",
        "dbtype": "sqlite3",
        "version": "10.1.0.1",
        "logtimezone": "UTC",
        "instanceid": "oc2fs70i9byd",
        "loglevel": "0",
        "csrf.disabled": true,
        "enable_previews": false,
        "sharing.federation.allowHttpFallback": true,
        "datadirectory": "\/home\/artur\/www\/owncloud-core\/data",
        "installed": true,
        "ldapIgnoreNamingRules": false,
        "htaccess.RewriteBase": "\/owncloud-core\/",
        "files_external_allow_create_new_local": "true"
    }
}
```

**List of activated apps:**

```
Enabled:
  - comments: 0.3.0
  - dav: 0.3.2
  - federatedfilesharing: 0.3.1
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.7.1
  - files_sharing: 0.10.1
  - files_texteditor: 2.2.1
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - testing: 0.1.0
  - updatenotification: 0.2.1
Disabled:
  - encryption
  - enterprise_key
  - files_antivirus
  - files_primary_s3
  - firewall
  - guests
  - multidirtest
  - user_management
  - user_ldap
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Logs
no log output

### PR DESCRIPTION
**Pull request by individual-it**
## Description
in case the `.htaccess` file cannot be updated throw an exception and show correct error message

## Related Issue
- Fixes #32227 

## Motivation and Context
I got stuck today during testing and it took me a while to figure out that the `.htaccess` file was not writeable, hope that fix will save time later

## How Has This Been Tested?
- try to install oC when `.htaccess` is not writeable 
- try to install oC when `.htaccess` is writeable 
- run `maintenance:update:htaccess` when `.htaccess` is not writeable 
- run `maintenance:update:htaccess` when `.htaccess` is writeable 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/lib/SetupTest.php [hunk 1]
```diff
@@ -133,4 +133,68 @@ public function testGetSupportedDatabaseException() {
...
+		if ($this->getCurrentUser() === 'root') {
+			$this->markTestSkipped(
+				'You are running tests as root - this test will not work in this case.'
+			);
+		}
+		$origServerRoot = \OC::$SERVERROOT;
+		$htaccessFile = \OC::$SERVERROOT . '/tests/data/.htaccess';
+		\touch($htaccessFile);
+		\chmod($htaccessFile, 0400);
+		\OC::$SERVERROOT = \OC::$SERVERROOT . '/tests/data';
```
Comment (skshetry) [inline tests/lib/SetupTest.php]: Why`OC::SERVERRROOT` is changed?

**Comment by individual-it** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by individual-it** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by PVince81** [comment]
@individual-it please finish if there is anything

@skshetry or confirm with an approval ?

**Comment by skshetry** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Review by phil-davis — APPROVED** [review]

**Comment by PVince81** [comment]
@individual-it please backport

**Comment by individual-it** [comment]
backport in #34461

**Comment by PVince81** [comment]
hmmm I think I misunderstood this PR...

all these years I think it was a given that some environments have a non-writable `.htaccess`, this is why we also have docs to tell the admins what to put there in such cases.

I might be wrong, or it might have changed...

@DeepDiver1975 do you remember ?

**Comment by PVince81** [comment]
> //suppress errors in case we don't have permissions for it

we might need to soften this and only display a warning instead of failing

**Comment by PVince81** [comment]
raised https://github.com/owncloud/core/issues/34471 for following up

**Pull request merged**